homeassistant/components/ecobee/climate.py

```diff
@@ -726,8 +734,11 @@ def set_hvac_mode(self, hvac_mode):
         if ecobee_value is None:
             _LOGGER.error("Invalid mode for set_hvac_mode: %s", hvac_mode)
             return
+
         self.data.ecobee.set_hvac_mode(self.thermostat_index, ecobee_value)
-        self.update_without_throttle = True
+
+        # set_hvac_mode was successful, reflect changes in thermostat
+        self.thermostat["settings"]["hvacMode"] = ecobee_value
```

Comment:
If you want to write the state optimistically, you'll need to call `self.schedule_update_ha_state()` or the entity won't be updated until the next poll

https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/entity.py#L635